api/src/opentrons/drivers/utils.py

```
@@ -29,96 +35,86 @@ def parse_string_value_from_substring(substring) -> str:
+            hold=parse_optional_number(data['H'], rounding_val)
+        )
+    except KeyError:
+        raise ParseError(f'Unexpected argument to'
+                         f' parse_lid_temperature_response: {temperature_string}')
 
 
 def parse_device_information(
-        device_info_string: str) -> Mapping[str, str]:
+        device_info_string: str) -> Dict[str, str]:
     """
-        Parse the modules's device information response.
+    Parse the modules's device information response.
 
-        Example response from temp-deck: "serial:aa11 model:bb22 version:cc33"
+    Example response from temp-deck: "serial:aa11 model:bb22 version:cc33"
     """
-    error_msg = 'Unexpected argument to parse_device_information: {}'.format(
-        device_info_string)
-    if not device_info_string or \
-            not isinstance(device_info_string, str):
-        raise ParseError(error_msg)
-    parsed_values = device_info_string.strip().split(' ')
-    if len(parsed_values) < 3:
-        log.error(error_msg)
-        raise ParseError(error_msg)
-    res = {
-        parse_key_from_substring(s): parse_string_value_from_substring(s)
-        for s in parsed_values[:3]
-    }
+    res = parse_key_values(device_info_string)
+
     for key in ['model', 'version', 'serial']:
         if key not in res:
-            raise ParseError(error_msg)
+            raise ParseError(f'Missing key {key} from device info '
+                             f'string {device_info_string}.')
     return res
 
 
+def parse_key_values(value: str) -> Dict[str, str]:
+    """Convert string in the format:
+        'key1:value1 key2:value2'
+        to dict
+        {'key1': 'value1', 'key2': 'value2'}
+    """
+    if not value or not isinstance(value, str):
+        raise ParseError(
+            f'Unexpected argument to parse_key_values: {value}'
+        )
+
+    res = {g.groupdict()['key']: g.groupdict()['value']
+           for g in KEY_VALUE_REGEX.finditer(value)}
+    return res
```

I don't know if we care, but in a string like

```
'a:b c d:e'
```

This will return `{'a':'b', 'd':'e'}`, silently ignoring `c`.

---

I don't care about it.